### ISSUE TITLE
Warning for translate triggers class function

### ISSUE THREAD

**Issue by thomasplevy** [WordPress-Coding-Standards/WordPress-Coding-Standards#719]
I have a class function in my plugin:

`
abstract class myclass { 
     public function translate( $key ) {}
}
`

This function doesn't do anything by default, but allows extending classes to pull a database value by $key and translate it. The extending classes ultimately run through some WP core i18n function like `__()`

During PHPCS runs this function triggers the i18nSniff error: The $text arg should be single a string literal, not "$key".

I see that some sniffs -- the "DontExtract" for example -- seem to allow this pattern, but this translate function doesn't follow the pattern

I tried to figure out how to rewrite the tests to allow for this exception to submit a PR but after some digging and attempts I decided I am not qualified to do so. Sorry...

Thanks,

**Comment by westonruter** [comment]
The problem in the i18nSniff is that it is not distinguishing between this example of defining a method and a function call for `translate()`. So the sniff needs to look at the context to skip if it is not a function call context. Probably the easiest way to check is to see if there is an opening brace following `translate()`, and if so, skip it.

**Comment by GaryJones** [comment]
If the code was _formatted_ as per WP coding standards, instead of all on one line, would the supposed (but incorrect) lack of i18n be flagged still?

**Comment by jrfnl** [comment]
@westonruter FYI: I'm working on a number of utility functions which I hope to introduce into WPCS soonish, one of these will be to determine if a `T_STRING` is a function call (rather than a function definition, method call etc). I think using that utility function near the top of the sniff to determine whether the function arguments should be inspected, will solve this issue.

**Comment by thomasplevy** [comment]
@GaryJones 

The function doesn't actually look like that anywhere in the codebase, just gave a simply one-liner as an example

The actual function in the abstract is:

`
public function translate( $key ) {
     return '';
}

**Comment by GaryJones** [comment]
Right - so does the same error still happen when it's formatted correctly?

I just want to ascertain that the issues is the _naming_ of your method that is causing the issue, rather than the way the code is formatted.

**Comment by thomasplevy** [comment]
@GaryJones -- is the above function formatted incorrectly?

**Comment by GaryJones** [comment]
```php
public function translate( $key ) {
     return '';
}
```

is correctly formatted, yes.

**Comment by westonruter** [comment]
```php
<?php
class Foo {
	public function translate( $key ) {
		return '';
	}
}
```

Yields:

```
--------------------------------------------------------------------------------------
FOUND 1 ERROR AND 1 WARNING AFFECTING 1 LINE
--------------------------------------------------------------------------------------
 3 | WARNING | Use of the "translate()" function is reserved for low-level API usage.
 3 | ERROR   | The $text arg should be single a string literal, not "$key".
```

This warning and error are bugs.
